File: Q50/Q50.ts
// Find all prime factors of a number. //

function primeFactors(n: number): number[] {
  const factors: number[] = [];
  while (n % 2 === 0) {
    factors.push(2);
    n = Math.floor(n / 2);
  }
  for (let i = 3; i <= Math.sqrt(n); i += 2) {
    while (n % i === 0) {
      factors.push(i);
      n = Math.floor(n / i);
    }
  }
  if (n > 2) {
    factors.push(n);
  }
  return factors;
}

console.log(primeFactors(56));

File: Q50/Q50.c
/*Prime factors*/
#include<stdio.h>
void primeFactors(int n);
void main() 
{
    int num;

    printf("Enter a positive integer: ");

    printf("Enter a positive integer :");

    scanf("%d",&num);

    if(num<=0)
    {
        printf("Please enter a positive integer\n");
    }
    else
    {
        printf("Prime factors of %d are :  ", num);
        
    }
    primeFactors(num);
  }
    void primeFactors(int n)
  {
    while (n%2==0)
    {
        printf("%d ",2);

        printf("Prime factors of %d are : ", n);
        
    }
    primeFactors(n);
}
